Makefile

```diff
@@ -403,6 +403,10 @@ docs/generate: | terraform/format
 		echo "Adding newline to the end of $(TFDOCS_FILE) file"; \
 		echo "" >> $(TFDOCS_FILE); \
 	fi
+	@if [ -n $(README_FILES) ] && [ "$$(tail -c1 modules/*/$(TFDOCS_FILE))" != "$('\n')" ]; then \
```

Comment:
Can you discuss how this works a bit? Biggest concern: when there are multiple modules, and one has a newline and one does not. Does this add a newline to both?

Reply:
I just tested this with two modules subfolders and it causes an ambiguous redirect error.  I'll have to work on this some more.  Good catch.